python/cuml/cuml/common/sparse_utils.py

```diff
@@ -36,8 +36,8 @@ def is_sparse(X):
     is_sparse : boolean
         is the input sparse?
     """
-    is_scipy_sparse = has_scipy() and scipy.sparse.isspmatrix(X)
-    return cupyx.scipy.sparse.isspmatrix(X) or is_scipy_sparse
+    is_scipy_sparse = has_scipy() and scipy.sparse.issparse(X)
```

Comment:
```suggestion
    is_scipy_sparse == has_scipy() and scipy.sparse.issparse(X)
```

Reply:
We are assigning a boolean value to the `is_scipy_sparse`variable, this should not be an equality check.